addons/mail/static/src/discuss/core/public_web/thread_actions.js

```diff
@@ -0,0 +1,32 @@
+import { threadActionsRegistry } from "@mail/core/common/thread_actions";
+import { SubChannelList } from "@mail/discuss/core/public_web/sub_channel_list";
+import { useChildSubEnv, useComponent } from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+import { usePopover } from "@web/core/popover/popover_hook";
+
+threadActionsRegistry.add("sub-channel", {
+    condition: (component) => component.thread?.hasSubThreadFeature,
+    icon: "fa fa-fw fa-rss",
+    name: _t("Show threads"),
+    component: SubChannelList,
+    setup(action) {
+        const component = useComponent();
+        if (!component.props.chatWindow) {
+            action.popover = usePopover(SubChannelList, {
+                onClose: () => action.close(),
+                fixedPosition: true,
+            });
+        }
+        useChildSubEnv({
+            subChannelMenu: {
+                open: () => action.open(),
+            },
+        });
+    },
+    open: (component, action) =>
+        action.popover?.open(component.root.el.querySelector(`[name="${action.id}"]`), {
+            thread: component.thread,
+        }),
+    sequence: (component) => (component.props.chatWindow ? 16 : 2),
+    toggle: true,
```

Comment:
Note that's there ongoing PR that improves thread actions, one of the improvements is ordering and grouping of thread actions: https://github.com/odoo/odoo/pull/178865

The current grouping of linked PR are loosely "viewers"/"actions"/"settings".

The difference between viewers and actions is quite subtle, generally "viewers" stay on same thread with rare redirections ("Attachments", "Members", "Pinned Messages"), whereas "actions" make change to the thread or the viewing of thread ("Invite People", "Search Messages", "Open in Discuss", "Rename Thread", etc.).

Although you named it "Show threads", I'd say this is more an "action" than a viewer. Reasoning is that most the time the screen is used to open the subthread, so that's an action. The popover design matches with this quick action aspect, and also it feels more natural to be next to invite people as a result.

Reply:
Thanks for the explanation, will re-order it accordingly if this PR gets merged first.